tests/system/fim/common_tasks/delete_files.py

```diff
@@ -45,11 +44,33 @@ def delete_files(input_file_path, n, output_file_path):
         to_delete = data
 
     # Delete the selected files
-    try:
-        for path in to_delete:
+    failed_deletions = []
+    for path in to_delete:
+        try:
             os.remove(path)
-    except Exception:
-        logging.error('Failed when deleting the selected files: ', exc_info=True)
+        except FileNotFoundError as e:
+            logging.error("File " + path + " not found.", exc_info=True)
+            raise e
+        except PermissionError:
+            logging.error("File " + path + " used by another process.", exc_info=True)
+            failed_deletions.append(path)
+            pass
+        except Exception:
+            raise Exception("Failed when deleting selected files")
+
+    # Retrying deletion on failed paths
+    for path in failed_deletions:
```

Comment:
I'd add some timeout here to wait for the agent to release the files.

Also you could refactor the logic to do 'delete attempts' every x seconds, kinda like the verify_json works.

Reply:
Added in 667b68e37e601b3e80e2d2a269bdd4433ba1b685